internal/simc.py

```diff
@@ -19,4 +25,107 @@ def sim_local(simc_path, profile_location, output_location, iterations):
+
+    base_path = "http://downloads.simulationcraft.org/nightly"
+
+    # Get filename of latest build of simc
+    try:
+        html = urlopen(f"{base_path}/?C=M;O=D").read().decode("utf-8")
+    except URLError:
+        print(f"Could not access download directory on simulationcraft.org")
+    # filename = re.search(r'<a href="(simc.+win64.+7z)">', html).group(1)
+    filename = list(
+        filter(None, re.findall(r'.+nonetwork.+|<a href="(simc.+win64.+7z)">', html))
+    )[0]
+    print(f"Latest simc: {filename}")
+
+    # Download latest build of simc
+    filepath = os.path.join(download_dir, filename)
+    if not os.path.exists(filepath):
+        url = f"{base_path}/{filename}"
+        print(f"Retrieving simc from url '{url}' to '{filepath}'.")
+        urlretrieve(url, filepath)
+    else:
+        print(f"Latest simc version already downloaded at {filename}.")
+        return filepath.strip(".7z")
+
+    # Unpack downloaded build and set simc_path
+    dir_name = filename[: filename.find(".7z")]
+    simc_path = os.path.join(download_dir, dir_name, "simc.exe")
+    if not os.path.exists(simc_path):
+        try:
+            cmd = f'{seven_zip_executable} x "{filepath}" -aoa -o"{download_dir}"'
+            print(f"Running unpack command '{cmd}'")
+            subprocess.call(cmd)
+
+            time.sleep(1)
+
+            # Nightly builds include their commit hash, we need to strip that out.
+            commit = dir_name.rsplit("-")[-1]
+            _rename_directory(f"{download_dir}/simc-*-win64/", commit)
+
+        except Exception as e:
+            print(f"Exception when unpacking: {e}")
+
+        # keep the latest 7z to remember current version, but clean up any other ones
+        _cleanup_older_files(download_dir, dir_name)
+
+    else:
+        print(f"Simc already exists at '{repr(simc_path)}'.")
+
+    return os.path.join(download_dir, dir_name)
+
+
+def _find_7zip(search_paths):
+    # Try to find 7zip, and raise an error if not
+    for executable in search_paths:
+        try:
+            if not os.path.exists(executable):
+                print(f"7Zip executable at '{executable}' does not exist.")
+                continue
+            else:
+                return executable
+                break
+        except:
+            continue
+    else:
+        raise RuntimeError(
+            "Could not unpack the auto downloaded SimulationCraft executable."
+            f"Please note that you need 7Zip installed at one of the following locations: {search_paths}."
+        )
+
+
+def _cleanup_older_files(download_dir, current_dir):
+    try:
+        files = glob.glob(f"{download_dir}/simc*")
+        for f in files:
+            if (
+                os.path.basename(f) != current_dir
+                and os.path.basename(f) != f"{current_dir}.7z"
+            ):
+                print(f"Removing old simc from '{os.path.basename(f)}'.")
+                os.remove(f)
+    except:
+        print(
```

Comment:
This fails unless you run the script as admin, which i didn't want to start asking people to do.
For now just telling them to manually clean it up.